internal/infrastructure/migration/v2.go

```diff
@@ -0,0 +1,167 @@
+// Package migration migrate current struct
+package migration
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-gormigrate/gormigrate/v2"
+	"github.com/gofrs/uuid"
+	"github.com/traPtitech/traPortfolio/internal/infrastructure/repository/model"
+	"gorm.io/gorm"
+)
+
+// v1 unique_index:idx_room_uniqueの削除
+func v2() *gormigrate.Migration {
+	return &gormigrate.Migration{
+		ID: "2",
+		Migrate: func(db *gorm.DB) error {
+			if err := db.AutoMigrate(&v2Project{}, &v2Contest{}, &v2ContestTeam{}); err != nil {
+				return err
+			}
+
+			// プロジェクト名の重複禁止
+			{
+				projects := make([]*model.Project, 0)
+				if err := db.Find(&projects).Error; err != nil {
+					return err
+				}
+
+				projectMap := make(map[string][]uuid.UUID, len(projects))
+				for _, p := range projects {
+					projectMap[p.Name] = append(projectMap[p.Name], p.ID)
+				}
+
+				updates := make(map[uuid.UUID]string, len(projects))
+				noDuplicate := false
+				for !noDuplicate {
+					noDuplicate = true
+					for name, arr := range projectMap {
```

Comment:
```suggestion
					for name, ids := range projectMap {
```
とか`projectIDs`とかが分かりやすそう